vuu-ui/packages/vuu-layout/test/typeOf.test.ts

```diff
@@ -0,0 +1,30 @@
+import { describe, expect, it } from "vitest";
+import { LayoutJSON, isLayoutJSON } from "../src";
+
+const validLayout: LayoutJSON = {
+  type: "Stack",
+};
+
+const invalidLayout: unknown = {
+  name: "invalid-layout-test",
+};
```

Comment:
What would happen if we had an object with all the required fields of a `LayoutJSON`, but isn't declared as such? Something like this:

```
const mayOrMayNotBeValidLayout: unknown = {
  type: "Stack",
};
```

Or if there's no type declaration at all?

Reply:
You will get a compile-time error if you give it an object whose type is declared as unknown ("Argument of type 'unknown' is not assignable to parameter of type 'LayoutJSON'") unless it gets cast as LayoutJSON. 
At run-time any object that has a "type" property will pass the validation

Reply:
object with no type won't give you a compile-time error as long as there is sufficient overlap between that object inferred type and LayoutJSON (meaning that that object has a "type" property)